### PR TITLE
Add support for editing guild widgets

### PR DESCRIPTION
**Pull request by MhmCats**
## Summary

This pull requests adds support for editing guild widgets.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/guild.py [hunk 1]
```diff
@@ -2353,6 +2353,30 @@ async def widget(self):
         data = await self._state.http.get_widget(self.id)
 
         return Widget(state=self._state, data=data)
+    
+    async def edit_widget(self, *, enabled: bool, channel: abc.Snowflake) -> None:
```
Comment (Zomatree) [inline discord/guild.py:2357]: Both parameters should be optional, also channel can be set to none.

File: discord/guild.py [hunk 2]
```diff
@@ -2353,6 +2353,30 @@ async def widget(self):
...
+
+        Edits the widget of the guild.
+
+        You must have the :attr:`~Permissions.manage_guild` permission to
+        use this
+
+        Parameters
+        -----------
+        enabled: :class:`bool`
+            Whether to enabled the widget for the guild.
```
Comment (SebbyLaw) [inline discord/guild.py:2368]:
```suggestion
            Whether to enable the widget for the guild.
```

**Review by SebbyLaw — CHANGES_REQUESTED** [review]

File: discord/guild.py [hunk 3]
```diff
@@ -2353,6 +2353,33 @@ async def widget(self):
         data = await self._state.http.get_widget(self.id)
 
         return Widget(state=self._state, data=data)
+    
+    async def edit_widget(self, *, enabled: bool = None, channel: abc.Snowflake = None) -> None:
```
Comment (Rapptz) [inline discord/guild.py:2357]:
```suggestion
    async def edit_widget(self, *, enabled: bool = utils.MISSING, channel: Optional[abc.Snowflake] = utils.MISSING) -> None:
```

File: discord/guild.py [hunk 4]
```diff
@@ -2353,6 +2353,33 @@ async def widget(self):
...
+        use this
+
+        .. versionadded:: 2.0
+
+        Parameters
+        -----------
+        enabled: :class:`bool`
+            Whether to enable the widget for the guild.
+        channel: :class:`abc.Snowflake`
+            The widget channel.
```
Comment (Rapptz) [inline discord/guild.py:2372]:
```suggestion
        channel: Optional[:class:`~discord.abc.Snowflake`]
            The new widget channel. ``None`` removes the widget channel.
```

File: discord/guild.py [hunk 5]
```diff
@@ -2353,6 +2353,33 @@ async def widget(self):
...
+
+        Raises
+        -------
+        Forbidden
+            You do not have permission to edit the widget.
+        HTTPException
+            Editing the widget failed.
+        """
+        channel = getattr(channel, 'id', None)
+        await self._state.http.edit_widget(self.id, enabled=enabled, channel_id=channel)
```
Comment (Rapptz) [inline discord/guild.py:2382]:
```suggestion
        payload = {}
        if channel is not utils.MISSING:
            payload['channel_id'] = None if channel is None else channel.id
        if enabled is not utils.MISSING:
            payload['enabled'] = enabled

        await self._state.http.edit_widget(self.id, payload=payload)
```

File: discord/http.py [hunk 6]
```diff
@@ -971,6 +971,9 @@ def get_audit_logs(self, guild_id, limit=100, before=None, after=None, user_id=N
 
     def get_widget(self, guild_id):
         return self.request(Route('GET', '/guilds/{guild_id}/widget.json', guild_id=guild_id))
+    
+    def edit_widget(self, guild_id, **payload):
```
Comment (Rapptz) [inline discord/http.py:975]:
```suggestion
    def edit_widget(self, guild_id, payload):
```

**Review by Rapptz — CHANGES_REQUESTED** [review]
This is inconsistent with the rest of the library.

**Pull request merged**